content/courses/advanced-ada/parts/resource_management/anonymous_access_types.rst

```diff
@@ -252,21 +254,553 @@ cover more specific details on anonymous access types.
+:ada:`Integer_Access` type, so that we can call the :ada:`Free` procedure to
+actually deallocate the object. Although this code compiles, it'll only work
+if both :ada:`access Integer` and :ada:`Integer_Access` types are using the
+same memory pool. Since we cannot really determine this, the result is
+potentially erroneous: it'll work if the compiler selected the same pool, but
+it'll fail otherwise.
+
+.. admonition:: Important
+
+    Because allocating memory for anonymous access types is potentially
+    dangerous, we can use the :ada:`No_Anonymous_Allocators` restriction
+    |mdash| which is available since Ada 2012 |mdash| to prevent this kind
+    of memory allocation being used in the code. For example:
+
+    .. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.No_Anonymous_Allocators
+        :class: ada-expect-compile-error
+
+        pragma Restrictions (No_Anonymous_Allocators);
+
+        procedure Show_Dangerous_Allocation is
+           IA : access Integer;
+        begin
+           IA := new Integer;
+           IA.all := 30;
+        end Show_Dangerous_Allocation;
+
+Possible solution using named access types
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A better solution to avoid issues when allocating and deallocating memory
+for anonymous access-to-object types is to allocate the object using a known
+pool. As mentioned before, the memory pool associated with a named access
+type is well-defined, so we can use this kind of types for memory allocation.
+In fact, we can use a named memory type to allocate an object via :ada:`new`,
+and then associate this allocated object with the access object of anonymous
+access type.
+
+Let's see a code example:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.Deallocation_Anonymous_Access_To_Object_1
+
+    with Ada.Unchecked_Deallocation;
+
+    procedure Show_Successful_Deallocation is
+
+       type Integer_Access is
+          access Integer;
+
+       procedure Free is
+         new Ada.Unchecked_Deallocation
+           (Object => Integer,
+            Name   => Integer_Access);
+
+       IA       : access Integer;
+       Typed_IA : Integer_Access;
+
+    begin
+       Typed_IA := new Integer;
+       IA := Typed_IA;
+       IA.all := 30;
+
+       --  Deallocation of the access object that has
+       --  an associated type:
+       Free (Typed_IA);
+
+    end Show_Successful_Deallocation;
+
+In this example, all operations related to memory allocation are exclusively
+making use of the :ada:`Integer_Access` type, which is a named access type.
+In fact, :ada:`new Integer` allocates the object in the pool associated with
+the :ada:`Integer_Access` type, and the call to :ada:`Free` deallocates this
+object from that pool. Therefore, associating this object with the :ada:`IA`
```

Comment:
Likewise, consider changing "object from that pool." to "object back into that pool."